src/jltypes.c

```diff
@@ -471,6 +472,37 @@ static jl_value_t *intersect_tuple(jl_tuple_t *a, jl_tuple_t *b,
         }
         jl_tupleset(tc, ci, ce);
     }
+    // Check for a length-constrained vararg
+    if (bseq) {
+        if (!jl_is_long(bn))
+            // set bn from eqc parameters
+            for (i = 0; i < eqc->n; i+=2)
+                if (eqc->data[i] == bn) {
+                    bn = eqc->data[i+1];
+                    break;
+                }
+        if (jl_is_long(bn)) {
+            long valen = jl_unbox_long(bn);
+            if (valen != (ai-bi+1))
+                result = (jl_value_t*)jl_bottom_type;
+        }
+        else if (jl_is_typevar(bn) && ((jl_tvar_t*)bn)->bound) {
+            // set eqc parameter from valen, to support func{N}(x...N)
+            int found = 0;
+            long valen = ai-bi+1;
+            for (i = 0; i < eqc->n; i+=2)
+                if (eqc->data[i] == bn) {
+                    eqc->data[i+1] = jl_box_long(valen);
```

Comment:
I'm not sure it's ok to overwrite the existing value here. Probably best to just call `extend`.